### PR TITLE
Add tests for count_downloads

### PR DESCRIPTION
**Pull request by M3rs**
Makes use of mocked get_release_info method to test basic counting, pattern matching, and verbose flags.

### CODE REVIEW

File: tests.py [hunk 1]
```diff
@@ -1,8 +1,110 @@
...
+                  'downloads': 2,
+                  'upload_time': datetime.date(2016, 10, 13)}
+
+    data = {'version': '1.0'}
+
+    yield [first_url, second_url], data
+
+
+# http://stackoverflow.com/q/21611559
+def Any(type):
```
Comment (hugovk) [inline tests.py]: Redefines builtin `type`

File: tests.py [hunk 2]
```diff
@@ -1,8 +1,110 @@
...
+                                       pattern='[Ff]ake')
+
+        self.assertEqual(count, 1)
+
+    @mock.patch('vanity.get_release_info', side_effect=single_release_info)
+    def test_verbose_calls_debug(self, get_release_func):
+        mock_logger = mock.Mock()
+        vanity.logger = mock_logger
+
+        count = vanity.count_downloads('fake package')
```
Comment (hugovk) [inline tests.py]: `count` is assigned to but never used, should it be asserted?

File: tests.py [hunk 3]
```diff
@@ -1,8 +1,110 @@
...
+
+        mock_logger.debug.assert_any_call(Any(str))
+
+    @mock.patch('vanity.get_release_info', side_effect=single_release_info)
+    def test_not_verbose_no_debug(self, get_release_func):
+        mock_logger = mock.Mock()
+        vanity.logger = mock_logger
+
+        count = vanity.count_downloads('fake package',
```
Comment (hugovk) [inline tests.py]: `count` is assigned to but never used, should it be asserted?

File: tests.py (hunk 1 above)
Comment (M3rs) [inline tests.py]: changed to object_type

File: tests.py (hunk 2 above)
Comment (M3rs) [inline tests.py]: Now asserted!

File: tests.py (hunk 3 above)
Comment (M3rs) [inline tests.py]: Now asserted!

**Pull request merged**